core/src/environment.rs

```diff
@@ -121,7 +121,7 @@ impl<K: Hash + Eq, V: PartialEq> Environment<K, V> {
     /// the most recent one. It uses this order, so calling `collect` on this iterator to create a
     /// hashmap would have the same values as the Environment. The element iterator type is `(&'env
     /// K, &'env V)`, with `'env` being the lifetime of the Environment.
-    pub fn iter_elems(&self) -> EnvElemIter<'_, K, V> {
+    pub fn iter_elems<'slf>(&'slf self) -> EnvElemIter<'slf, K, V> {
```

Comment:
Yes, and I doubt lifetime elision rules will change, so this is perfectly safe. The change was only to make it more explicit, since it's very important to what is happening and not necessarily intuitive. I don't have strong opinions about it though; I'm happy to revert this change.

Reply:
I see, thanks. I just wondered if there was something else to it. At any rate, it's not too verbose, and it's more explicit, so it probably doesn't hurt

Reply:
Ah. However, clippy doesn't like it. I've put in `#[allow(clippy::needless_lifetimes)]`, but let me know if you want me to just revert the change.

Reply:
Without clippy, I would have said go for it, but now I wonder if it's not just simpler and more idiomatic to just ditch the explicit lifetime, yeah.

Reply:
alright, reverted.